### ISSUE TITLE
SCons Numpy dependency

### ISSUE THREAD

**Issue by oxidian24** [precice/precice#114]
I have read your instruction to compile preCICE, but I did not found that numpy must be installed.
I got an error when I tried to start compilation. The error says ```np not defined```. I found, that the paths where the packages for python should be installed are empty and I installed it. But it took me about two hours to find out. You could mention that numpy is needed for python.

My suggestion is to add information.

```
python
import sys
print '\n'.join(sys.path)
exit()
```

To get the paths and check if numpy is installed.

Then

```
sudo apt-get install python-pip  
sudo pip install numpy
```

for Ubuntu.

**Comment by BenjaminRueth** [comment]
@oxidian24 Thank you for opening this issue. We will try to improve this situation.

**Comment by BenjaminRueth** [comment]
@oxidian24 I am not sure what kind of error message/information you are proposing. You wrote:

```
sudo apt-get install python-pip  
sudo pip install numpy
```
Do you propose to give a hint on how to install python/numpy?

I also formatted your issue using \```<your code goes here>\```. Without this formatting it is hard to read. I hope I did not format it wrong.

**Comment by MakisH** [comment]
Hey @oxidian24,

for the missing information in the building instructions, I am to blame. It got lost while rewriting the building instructions a few days ago. I added it now.

You don't need to install pip in order to install numpy on Ubuntu. There is already the package `python-numpy` in the Ubuntu repositories.

**Comment by oxidian24** [comment]
I am not python programmer or even best unix user.

1st PROBLEM:

Please see the file SConstruct
There is this:
`
try:
    import numpy as np
except ImportError:
    pass
`
So IMHO the compilation need numpy. I did not find out how to specify where it is. So I have installed it using help/stackoverflow answer which I have found using google.

And the problem I had was on line 249:
`numpyIncPathDefault = np.get_include()`
where np was undefined. So I was looking for solution and I found out that np was not installed or the path was not defined. So that was the way I solved the problem.

2nd PROBLEM:

Another thing which you could mention is to add the PYTHONPATH to .rcbash
`export PYTHONPATH="/usr/include/mpi"`
(and restart linux)
Because in my case the compiler (or scons?) could not find the mpi. 
But now I see it still does not recognise the header. I checked that python's sys.path is OK. So I cannot find solution for this problem.

**Comment by oxidian24** [comment]
Yesterday I was told that when I run the compilator on more cores I can run into a problem due to not enoth memory problem. This is another point which was not in the instructions. I have several tabs open now with the instructions but I don't see Requirements section. Saying how much memory the compilation need is one of the basic things. At the moment I cannot compile the program. I uninstalled mpi and installed again, and then I have started the compilation with 1 core with mpicxx compiler. This always leads to symptoms similar to "not enough memory" - however I cannot confirm that.  In Ubuntu 17.10 I had have monitoring with GUI opened ... however the screen gets black when I start the compilation, so I cannot see any text. 

Another good point would be to add instruction how to run memory monitoring in a shell or how to check if there is enough memory.

Note, that on my Machine I used to compile programs with G++ on Windows, but now I am running Virtual Machine with 2241 MB and 7 cored assigned. I cannot assign more because I have 3GB total on Windows XP. So when I start the compilation on Ubuntu in VM, the system is paused by VirtualBox.

**Comment by oxidian24** [comment]
Yesterday I was told that when I run the compilator on more cores I can run into a problem due to not enoth memory problem. This is another point which was not in the instructions. I have several tabs open now with the instructions but I don't see Requirements section. Saying how much memory the compilation need is one of the basic things. At the moment I cannot compile the program. I uninstalled mpi and installed again, and then I have started the compilation with 1 core with mpicxx compiler. This always leads to symptoms similar to "not enough memory" - however I cannot confirm that.  In Ubuntu 17.10 I had have monitoring with GUI opened ... however the screen gets black when I start the compilation, so I cannot see any text. 

Another good point would be to add instruction how to run memory monitoring in a shell or how to check if there is enough memory.

Note, that on my Machine I used to compile programs with G++ on Windows, but now I am running Virtual Machine with 2241 MB and 7 cored assigned. I cannot assign more because I have 3GB total on Windows XP. So when I start the compilation on Ubuntu in VM, the system is paused by VirtualBox.

Now I got error MPI not found again and when I type sudo apt install mpi I got this response: 
liblam4 latest version (7.1.4-3.1) is installed
